pkg/apiserver/resthandler.go

```diff
@@ -470,11 +470,11 @@ func PatchResource(r rest.Patcher, scope RequestScope, typer runtime.ObjectTyper
 		ctx := scope.ContextFunc(req)
 		ctx = api.WithNamespace(ctx, namespace)
 
-		// PATCH requires same permission as UPDATE
-		if admit.Handles(admission.Update) {
+		if admit != nil && admit.Handles(admission.Patch) {
 			userInfo, _ := api.UserFrom(ctx)
 
-			err = admit.Admit(admission.NewAttributesRecord(obj, scope.Kind.GroupKind(), namespace, name, scope.Resource.GroupResource(), scope.Subresource, admission.Update, userInfo))
+			// TODO this record should include the patch itself.  Right now we can't do that because patches are not runtime.Objects
```

Comment:
> Can you give an example of when you would want to use this?

With or without an old/new combination for update based admission?  Obviously, if you have old and new, you can build the logical patch yourself to make a determination.  The admission flow doesn't currently do that and it's a more significant change.  I don't disagree that would cleaner, but this is a critical security vulnerability now (bypasses all field level authorization of updates) and I was hoping avoid a full refactor.